pyasdf/asdftypes.py

```diff
@@ -108,13 +109,37 @@ def from_yaml_tag(self, tag):
 
     @lru_cache(5)
     def has_hook(self, hook_name):
+        """
+        Returns `True` if the given hook name exists on any of the managed
+        types.
+        """
         for cls in self._all_types:
             if hasattr(cls, hook_name):
                 return True
         return False
```

Comment:
The hooks are run during various phases of saving and loading.  (pre_write, post_write, post_read, etc.)  Certain types (mainly ndarray) need to be able to do things at those phases.  For example, ndarray needs to determine all of the underlying data blocks that are in use through the whole tree, so that unused blocks are garbage collected, duplicate blocks are removed etc.  So it doesn't have much to do with validation -- though we could add pre and post validation hooks someday if there were a need for it that came up.